pkg/plugins/execplugin.go

```diff
@@ -223,3 +230,28 @@ func (p *ExecPlugin) updateResMapValues(output []byte, rm resmap.ResMap) error {
 	}
 	return nil
 }
+
+// updateResourceOptions updates the generator options for each resource in the
+// given ResMap based on plugin provided annotations.
+func (p *ExecPlugin) updateResourceOptions(rm resmap.ResMap) resmap.ResMap {
+	for _, r := range rm.Resources() {
+		disableHash := true
+		// Disable name hashing by default and require plugin to explicitly
+		// request it for each resource.
+		annotations := r.GetAnnotations()
+		behavior := annotations[behaviorAnnotation]
+		if _, ok := annotations[hashAnnotation]; ok {
+			disableHash = false
+		}
```

Comment:
I would have expected to be able to specify `kustomize.config.k8s.io/needs-hash: "false"`, but the code seems to enable hashing by just the presence of the annotation.

Reply:
Yes this was a deliberate choice and in hind sight the example I put in of `"true"` does make the expectation of being able to set "false" seem logical.

The reason I opted for a simple existence check on the annotation vs treating it as a bool was to avoid issues with YAML bool handling for a string field or a possible philosophical debate about what values should be considered as true vs false when interpreting the string value of the annotation. While my initial reaction is to fix the example to make this feel less like a "bool" I am open to the alternative if that's what others feel is more intuitive.

Reply:
I've grown accustomed to the `true` / `"true"` problem with YAML, it already in issue with environment variables, labels and other annotations. The Nginx ingress controller, as an example, uses [annotations](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/) that can be `"true"` or `"false"`.

That said, and although I have not seen any other application do this, when it is documented so clearly as you have done in the latest commit I have no problem using just the presence of the annotation.

So... that's a "-0" I guess. :-)

Reply:
Thinking about this over the last two days and now I actually have changed my mind on this and have updated the PR to handle the annotation as a bool.

> although I have not seen any other application do this

While I was thinking in terms of "Exists" / "DoesNotExist" similar to nodeAffinity rules where the value of the key being matched doesn't actually matter. This point that other applications are handling similar situations with true/false strings made me reconsider. Despite the existence check being sufficient for this use case I think being consistent with other ecosystem tools is more important.